lib/reader.js

```diff
@@ -1,115 +1,59 @@
 'use strict';
 
-const JSONStream = require('JSONStream');
 const pack = require('browser-pack');
-const mergeStream = require('merge-stream');
-const stream = require('readable-stream');
-
-// Ensure output order is the same as input order. The sorter only sorts the
-// modules, not execution order. `order` is consumed by browserify's
-// `browser-pack`. The feeds already have an order, but they are always `0` as
-// each feed only have a single entrypoint.
-class OrderDecorator extends stream.Transform {
-    constructor (order) {
-        super({
-            objectMode: true,
-        });
-
-        this.order = order;
-    }
-
-    _transform (row, encoding, next) {
-        if (row.entry === true) {
-            row.order = this.order;
-        }
-        next(null, row);
-    }
+const getStream = require('get-stream');
+const assert = require('assert');
+
+function getRows (feeds) {
+    const existMap = new Map();
+    return feeds
+        .reduce((acc, feed, order) => {
+            feed.forEach((row) => {
+                const copy = { ...row };
+                // add browserify.order and flatten
+                delete copy.order;
+                if (copy.entry === true) {
+                    copy.order = order;
+                }
+                acc.push(copy);
+            });
+            return acc;
+        }, [])
+        .filter(row => {
+            assert(row, `Expected feed entry to be an object, got "${typeof row}"`);
+            assert(row.id, `Expected feed entry to have an id, got "${row.id}"`);
+            if (existMap.has(row.id)) {
+                // FIXME: This is too simplistic. We have to handle the case of rows
+                // having keys like `expose` set, and ensure the correct configuration
+                // bubbles up
+                return false;
+            }
+            existMap.set(row.id, true);
+            return true;
+        })
+        .sort((a, b) => a.id.localeCompare(b.id));
 }
 
-class SortAndDedupe extends stream.Transform {
-    constructor () {
-        super({
-            objectMode: true,
-        });
-
-        this.rows = new Map();
-    }
-
-    _transform (row, encoding, next) {
-        // FIXME: This is too simplistic. We have to handle the case of rows
-        // having keys like `expose` set, and ensure the correct configuration
-        // bubbles up
-        if (row && row.id && !this.rows.has(row.id)) {
-            this.rows.set(row.id, row);
-        }
-        next();
-    }
 
-    _flush (next) {
-        Array
-            .from(this.rows.values())
-            .sort(compareId)
-            .forEach(row => this.push(row));
-        next();
-    }
-}
+module.exports = async (feeds) => {
+    assert(Array.isArray(feeds), `Expected an array of feed arrays, got "${feeds}"`);
+    assert(feeds.length > 0, 'Expected at least 1 feed');
+    assert(feeds[0].length > 0, 'Expected at least 1 feed with more than one entry');
```

Comment:
This isn't checking what the message says its checking. You're expecting that the first item has at least 1 item. What about using `every` or `some`? 

```js
assert(feeds.some(feed => !!feed.length), `...`)
```

Reply:
Also, could we check the heck out of the feeds here so that we don't need to perform assertions inside the getRows filter on line 23? Maybe we should be defining a detailed schema with joi and using that to validate input on line 39?